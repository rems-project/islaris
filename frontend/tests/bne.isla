(trace
  (declare-const v3 (_ BitVec 1))
  (declare-const v4 (_ BitVec 2))
  (declare-const v5 (_ BitVec 1))
  (declare-const v6 (_ BitVec 1))
  (declare-const v7 (_ BitVec 1))
  (declare-const v8 (_ BitVec 1))
  (declare-const v10 (_ BitVec 1))
  (declare-const v11 (_ BitVec 4))
  (declare-const v12 (_ BitVec 1))
  (declare-const v13 (_ BitVec 1))
  (declare-const v14 (_ BitVec 8))
  (declare-const v15 (_ BitVec 1))
  (declare-const v16 (_ BitVec 5))
  (declare-const v17 (_ BitVec 1))
  (declare-const v18 (_ BitVec 1))
  (declare-const v19 (_ BitVec 1))
  (declare-const v21 (_ BitVec 1))
  (declare-const v22 (_ BitVec 1))
  (declare-const v23 (_ BitVec 1))
  (declare-const v24 (_ BitVec 1))
  (declare-const v25 (_ BitVec 1))
  (declare-const v26 (_ BitVec 1))
  (declare-const v27 (_ BitVec 1))
  (declare-const v28 (_ BitVec 1))
  (cycle)
  (read-reg |SEE| nil (_ bv-1 128))
  (write-reg |SEE| nil (_ bv1498 128))
  (write-reg |__unconditional| nil true)
  (read-reg |__v85_implemented| nil false)
  (read-reg |PSTATE| ((_ field |Z|)) (_ struct (|M| v16) (|E| v8) (|DIT| v7) (|BTYPE| v4) (|J| v15) (|I| v12) (|SP| #b1) (|D| v6) (|TCO| v24) (|Z| v27) (|A| v3) (|IL| v13) (|EL| #b10) (|IT| v14) (|SSBS| v22) (|UAO| v25) (|T| v23) (|nRW| v28) (|N| v17) (|SS| v21) (|Q| v19) (|GE| v11) (|V| v26) (|PAN| v18) (|C| v5) (|F| v10)))
  (define-const v38 (not (= v27 #b1)))
  (branch 0 "model/aarch64.sail 12137:4 - 12139:5")
  (assert v38)
  (declare-const v39 (_ BitVec 64))
  (read-reg |_PC| nil v39)
  (define-const v40 (bvadd v39 #xfffffffffffff3f0))
  (read-reg |PSTATE| ((_ field |nRW|)) (_ struct (|M| v16) (|E| v8) (|DIT| v7) (|BTYPE| v4) (|J| v15) (|I| v12) (|SP| #b1) (|D| v6) (|TCO| v24) (|Z| v27) (|A| v3) (|IL| v13) (|EL| #b10) (|IT| v14) (|SSBS| v22) (|UAO| v25) (|T| v23) (|nRW| v28) (|N| v17) (|SS| v21) (|Q| v19) (|GE| v11) (|V| v26) (|PAN| v18) (|C| v5) (|F| v10)))
  (define-const v52 (= v28 #b1))
  (read-reg |CFG_ID_AA64PFR0_EL1_EL0| nil #x1)
  (read-reg |CFG_ID_AA64PFR0_EL1_EL1| nil #x1)
  (read-reg |CFG_ID_AA64PFR0_EL1_EL2| nil #x1)
  (read-reg |CFG_ID_AA64PFR0_EL1_EL3| nil #x1)
  (assert (not v52))
  (read-reg |__highest_el_aarch32| nil false)
  (assert (not v52))
  (define-const v55 v40)
  (branch-address v55)
  (define-const v56 v40)
  (read-reg |PSTATE| ((_ field |nRW|)) (_ struct (|M| v16) (|E| v8) (|DIT| v7) (|BTYPE| v4) (|J| v15) (|I| v12) (|SP| #b1) (|D| v6) (|TCO| v24) (|Z| v27) (|A| v3) (|IL| v13) (|EL| #b10) (|IT| v14) (|SSBS| v22) (|UAO| v25) (|T| v23) (|nRW| v28) (|N| v17) (|SS| v21) (|Q| v19) (|GE| v11) (|V| v26) (|PAN| v18) (|C| v5) (|F| v10)))
  (define-const v57 (= v28 #b1))
  (read-reg |CFG_ID_AA64PFR0_EL1_EL0| nil #x1)
  (read-reg |CFG_ID_AA64PFR0_EL1_EL1| nil #x1)
  (read-reg |CFG_ID_AA64PFR0_EL1_EL2| nil #x1)
  (read-reg |CFG_ID_AA64PFR0_EL1_EL3| nil #x1)
  (assert (not v57))
  (read-reg |__highest_el_aarch32| nil false)
  (assert (not v57))
  (read-reg |PSTATE| ((_ field |EL|)) (_ struct (|M| v16) (|E| v8) (|DIT| v7) (|BTYPE| v4) (|J| v15) (|I| v12) (|SP| #b1) (|D| v6) (|TCO| v24) (|Z| v27) (|A| v3) (|IL| v13) (|EL| #b10) (|IT| v14) (|SSBS| v22) (|UAO| v25) (|T| v23) (|nRW| v28) (|N| v17) (|SS| v21) (|Q| v19) (|GE| v11) (|V| v26) (|PAN| v18) (|C| v5) (|F| v10)))
  (read-reg |CFG_ID_AA64PFR0_EL1_EL2| nil #x1)
  (read-reg |CFG_ID_AA64PFR0_EL1_EL3| nil #x1)
  (read-reg |CFG_ID_AA64PFR0_EL1_EL3| nil #x1)
  (read-reg |__highest_el_aarch32| nil false)
  (read-reg |SCR_EL3| nil #x00000000)
  (read-reg |CFG_ID_AA64PFR0_EL1_EL2| nil #x1)
  (read-reg |CFG_ID_AA64PFR0_EL1_EL0| nil #x1)
  (read-reg |CFG_ID_AA64PFR0_EL1_EL1| nil #x1)
  (read-reg |CFG_ID_AA64PFR0_EL1_EL2| nil #x1)
  (read-reg |CFG_ID_AA64PFR0_EL1_EL3| nil #x1)
  (read-reg |__v81_implemented| nil true)
  (read-reg |__v84_implemented| nil false)
  (read-reg |CFG_ID_AA64PFR0_EL1_EL3| nil #x1)
  (read-reg |CFG_ID_AA64PFR0_EL1_EL3| nil #x1)
  (read-reg |__highest_el_aarch32| nil false)
  (read-reg |SCR_EL3| nil #x00000000)
  (read-reg |TCR_EL2| nil #x0000000000000000)
  (read-reg |__v83_implemented| nil false)
  (write-reg |_PC| nil v56)
  (write-reg |__PC_changed| nil true))
(trace
  (declare-const v3 (_ BitVec 1))
  (declare-const v4 (_ BitVec 2))
  (declare-const v5 (_ BitVec 1))
  (declare-const v6 (_ BitVec 1))
  (declare-const v7 (_ BitVec 1))
  (declare-const v8 (_ BitVec 1))
  (declare-const v10 (_ BitVec 1))
  (declare-const v11 (_ BitVec 4))
  (declare-const v12 (_ BitVec 1))
  (declare-const v13 (_ BitVec 1))
  (declare-const v14 (_ BitVec 8))
  (declare-const v15 (_ BitVec 1))
  (declare-const v16 (_ BitVec 5))
  (declare-const v17 (_ BitVec 1))
  (declare-const v18 (_ BitVec 1))
  (declare-const v19 (_ BitVec 1))
  (declare-const v21 (_ BitVec 1))
  (declare-const v22 (_ BitVec 1))
  (declare-const v23 (_ BitVec 1))
  (declare-const v24 (_ BitVec 1))
  (declare-const v25 (_ BitVec 1))
  (declare-const v26 (_ BitVec 1))
  (declare-const v27 (_ BitVec 1))
  (declare-const v28 (_ BitVec 1))
  (cycle)
  (read-reg |SEE| nil (_ bv-1 128))
  (write-reg |SEE| nil (_ bv1498 128))
  (write-reg |__unconditional| nil true)
  (read-reg |__v85_implemented| nil false)
  (read-reg |PSTATE| ((_ field |Z|)) (_ struct (|M| v16) (|E| v8) (|DIT| v7) (|BTYPE| v4) (|J| v15) (|I| v12) (|SP| #b1) (|D| v6) (|TCO| v24) (|Z| v27) (|A| v3) (|IL| v13) (|EL| #b10) (|IT| v14) (|SSBS| v22) (|UAO| v25) (|T| v23) (|nRW| v28) (|N| v17) (|SS| v21) (|Q| v19) (|GE| v11) (|V| v26) (|PAN| v18) (|C| v5) (|F| v10)))
  (define-const v38 (not (= v27 #b1)))
  (branch 0 "model/aarch64.sail 12137:4 - 12139:5")
  (assert (not v38)))
