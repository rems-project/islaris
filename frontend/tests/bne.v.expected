From isla Require Import isla_lang.

Definition trace : list trc := [
  [
    Smt (DeclareConst 3%Z (Ty_BitVec 1%N)) Mk_annot;
    Smt (DeclareConst 4%Z (Ty_BitVec 2%N)) Mk_annot;
    Smt (DeclareConst 5%Z (Ty_BitVec 1%N)) Mk_annot;
    Smt (DeclareConst 6%Z (Ty_BitVec 1%N)) Mk_annot;
    Smt (DeclareConst 7%Z (Ty_BitVec 1%N)) Mk_annot;
    Smt (DeclareConst 8%Z (Ty_BitVec 1%N)) Mk_annot;
    Smt (DeclareConst 10%Z (Ty_BitVec 1%N)) Mk_annot;
    Smt (DeclareConst 11%Z (Ty_BitVec 4%N)) Mk_annot;
    Smt (DeclareConst 12%Z (Ty_BitVec 1%N)) Mk_annot;
    Smt (DeclareConst 13%Z (Ty_BitVec 1%N)) Mk_annot;
    Smt (DeclareConst 14%Z (Ty_BitVec 8%N)) Mk_annot;
    Smt (DeclareConst 15%Z (Ty_BitVec 1%N)) Mk_annot;
    Smt (DeclareConst 16%Z (Ty_BitVec 5%N)) Mk_annot;
    Smt (DeclareConst 17%Z (Ty_BitVec 1%N)) Mk_annot;
    Smt (DeclareConst 18%Z (Ty_BitVec 1%N)) Mk_annot;
    Smt (DeclareConst 19%Z (Ty_BitVec 1%N)) Mk_annot;
    Smt (DeclareConst 21%Z (Ty_BitVec 1%N)) Mk_annot;
    Smt (DeclareConst 22%Z (Ty_BitVec 1%N)) Mk_annot;
    Smt (DeclareConst 23%Z (Ty_BitVec 1%N)) Mk_annot;
    Smt (DeclareConst 24%Z (Ty_BitVec 1%N)) Mk_annot;
    Smt (DeclareConst 25%Z (Ty_BitVec 1%N)) Mk_annot;
    Smt (DeclareConst 26%Z (Ty_BitVec 1%N)) Mk_annot;
    Smt (DeclareConst 27%Z (Ty_BitVec 1%N)) Mk_annot;
    Smt (DeclareConst 28%Z (Ty_BitVec 1%N)) Mk_annot;
    Cycle Mk_annot;
    ReadReg "SEE" [] (Val_I (-1)%Z 128%Z) Mk_annot;
    WriteReg "SEE" [] (Val_I 1498%Z 128%Z) Mk_annot;
    WriteReg "__unconditional" [] (Val_Bool true) Mk_annot;
    ReadReg "__v85_implemented" [] (Val_Bool false) Mk_annot;
    ReadReg "PSTATE" [Field "Z"] (Val_Struct [("M", Val_Symbolic 16%Z); ("E", Val_Symbolic 8%Z); ("DIT", Val_Symbolic 7%Z); ("BTYPE", Val_Symbolic 4%Z); ("J", Val_Symbolic 15%Z); ("I", Val_Symbolic 12%Z); ("SP", Val_Bits [BV{1%N} 0x1%Z]); ("D", Val_Symbolic 6%Z); ("TCO", Val_Symbolic 24%Z); ("Z", Val_Symbolic 27%Z); ("A", Val_Symbolic 3%Z); ("IL", Val_Symbolic 13%Z); ("EL", Val_Bits [BV{2%N} 0x2%Z]); ("IT", Val_Symbolic 14%Z); ("SSBS", Val_Symbolic 22%Z); ("UAO", Val_Symbolic 25%Z); ("T", Val_Symbolic 23%Z); ("nRW", Val_Symbolic 28%Z); ("N", Val_Symbolic 17%Z); ("SS", Val_Symbolic 21%Z); ("Q", Val_Symbolic 19%Z); ("GE", Val_Symbolic 11%Z); ("V", Val_Symbolic 26%Z); ("PAN", Val_Symbolic 18%Z); ("C", Val_Symbolic 5%Z); ("F", Val_Symbolic 10%Z)]) Mk_annot;
    Smt (DefineConst 38%Z (Unop (Not) (Binop (Eq) (Val (Val_Symbolic 27%Z) Mk_annot) (Val (Val_Bits [BV{1%N} 0x1%Z]) Mk_annot) Mk_annot) Mk_annot)) Mk_annot;
    Branch 0%Z "model/aarch64.sail 12137:4 - 12139:5" Mk_annot;
    Smt (Assert (Val (Val_Symbolic 38%Z) Mk_annot)) Mk_annot;
    Smt (DeclareConst 39%Z (Ty_BitVec 64%N)) Mk_annot;
    ReadReg "_PC" [] (Val_Symbolic 39%Z) Mk_annot;
    Smt (DefineConst 40%Z (Manyop (Bvmanyarith Bvadd) [Val (Val_Symbolic 39%Z) Mk_annot; Val (Val_Bits [BV{64%N} 0xfffffffffffff3f0%Z]) Mk_annot] Mk_annot)) Mk_annot;
    ReadReg "PSTATE" [Field "nRW"] (Val_Struct [("M", Val_Symbolic 16%Z); ("E", Val_Symbolic 8%Z); ("DIT", Val_Symbolic 7%Z); ("BTYPE", Val_Symbolic 4%Z); ("J", Val_Symbolic 15%Z); ("I", Val_Symbolic 12%Z); ("SP", Val_Bits [BV{1%N} 0x1%Z]); ("D", Val_Symbolic 6%Z); ("TCO", Val_Symbolic 24%Z); ("Z", Val_Symbolic 27%Z); ("A", Val_Symbolic 3%Z); ("IL", Val_Symbolic 13%Z); ("EL", Val_Bits [BV{2%N} 0x2%Z]); ("IT", Val_Symbolic 14%Z); ("SSBS", Val_Symbolic 22%Z); ("UAO", Val_Symbolic 25%Z); ("T", Val_Symbolic 23%Z); ("nRW", Val_Symbolic 28%Z); ("N", Val_Symbolic 17%Z); ("SS", Val_Symbolic 21%Z); ("Q", Val_Symbolic 19%Z); ("GE", Val_Symbolic 11%Z); ("V", Val_Symbolic 26%Z); ("PAN", Val_Symbolic 18%Z); ("C", Val_Symbolic 5%Z); ("F", Val_Symbolic 10%Z)]) Mk_annot;
    Smt (DefineConst 52%Z (Binop (Eq) (Val (Val_Symbolic 28%Z) Mk_annot) (Val (Val_Bits [BV{1%N} 0x1%Z]) Mk_annot) Mk_annot)) Mk_annot;
    ReadReg "CFG_ID_AA64PFR0_EL1_EL0" [] (Val_Bits [BV{4%N} 0x1%Z]) Mk_annot;
    ReadReg "CFG_ID_AA64PFR0_EL1_EL1" [] (Val_Bits [BV{4%N} 0x1%Z]) Mk_annot;
    ReadReg "CFG_ID_AA64PFR0_EL1_EL2" [] (Val_Bits [BV{4%N} 0x1%Z]) Mk_annot;
    ReadReg "CFG_ID_AA64PFR0_EL1_EL3" [] (Val_Bits [BV{4%N} 0x1%Z]) Mk_annot;
    Smt (Assert (Unop (Not) (Val (Val_Symbolic 52%Z) Mk_annot) Mk_annot)) Mk_annot;
    ReadReg "__highest_el_aarch32" [] (Val_Bool false) Mk_annot;
    Smt (Assert (Unop (Not) (Val (Val_Symbolic 52%Z) Mk_annot) Mk_annot)) Mk_annot;
    Smt (DefineConst 55%Z (Val (Val_Symbolic 40%Z) Mk_annot)) Mk_annot;
    BranchAddress (Val_Symbolic 55%Z) Mk_annot;
    Smt (DefineConst 56%Z (Val (Val_Symbolic 40%Z) Mk_annot)) Mk_annot;
    ReadReg "PSTATE" [Field "nRW"] (Val_Struct [("M", Val_Symbolic 16%Z); ("E", Val_Symbolic 8%Z); ("DIT", Val_Symbolic 7%Z); ("BTYPE", Val_Symbolic 4%Z); ("J", Val_Symbolic 15%Z); ("I", Val_Symbolic 12%Z); ("SP", Val_Bits [BV{1%N} 0x1%Z]); ("D", Val_Symbolic 6%Z); ("TCO", Val_Symbolic 24%Z); ("Z", Val_Symbolic 27%Z); ("A", Val_Symbolic 3%Z); ("IL", Val_Symbolic 13%Z); ("EL", Val_Bits [BV{2%N} 0x2%Z]); ("IT", Val_Symbolic 14%Z); ("SSBS", Val_Symbolic 22%Z); ("UAO", Val_Symbolic 25%Z); ("T", Val_Symbolic 23%Z); ("nRW", Val_Symbolic 28%Z); ("N", Val_Symbolic 17%Z); ("SS", Val_Symbolic 21%Z); ("Q", Val_Symbolic 19%Z); ("GE", Val_Symbolic 11%Z); ("V", Val_Symbolic 26%Z); ("PAN", Val_Symbolic 18%Z); ("C", Val_Symbolic 5%Z); ("F", Val_Symbolic 10%Z)]) Mk_annot;
    Smt (DefineConst 57%Z (Binop (Eq) (Val (Val_Symbolic 28%Z) Mk_annot) (Val (Val_Bits [BV{1%N} 0x1%Z]) Mk_annot) Mk_annot)) Mk_annot;
    ReadReg "CFG_ID_AA64PFR0_EL1_EL0" [] (Val_Bits [BV{4%N} 0x1%Z]) Mk_annot;
    ReadReg "CFG_ID_AA64PFR0_EL1_EL1" [] (Val_Bits [BV{4%N} 0x1%Z]) Mk_annot;
    ReadReg "CFG_ID_AA64PFR0_EL1_EL2" [] (Val_Bits [BV{4%N} 0x1%Z]) Mk_annot;
    ReadReg "CFG_ID_AA64PFR0_EL1_EL3" [] (Val_Bits [BV{4%N} 0x1%Z]) Mk_annot;
    Smt (Assert (Unop (Not) (Val (Val_Symbolic 57%Z) Mk_annot) Mk_annot)) Mk_annot;
    ReadReg "__highest_el_aarch32" [] (Val_Bool false) Mk_annot;
    Smt (Assert (Unop (Not) (Val (Val_Symbolic 57%Z) Mk_annot) Mk_annot)) Mk_annot;
    ReadReg "PSTATE" [Field "EL"] (Val_Struct [("M", Val_Symbolic 16%Z); ("E", Val_Symbolic 8%Z); ("DIT", Val_Symbolic 7%Z); ("BTYPE", Val_Symbolic 4%Z); ("J", Val_Symbolic 15%Z); ("I", Val_Symbolic 12%Z); ("SP", Val_Bits [BV{1%N} 0x1%Z]); ("D", Val_Symbolic 6%Z); ("TCO", Val_Symbolic 24%Z); ("Z", Val_Symbolic 27%Z); ("A", Val_Symbolic 3%Z); ("IL", Val_Symbolic 13%Z); ("EL", Val_Bits [BV{2%N} 0x2%Z]); ("IT", Val_Symbolic 14%Z); ("SSBS", Val_Symbolic 22%Z); ("UAO", Val_Symbolic 25%Z); ("T", Val_Symbolic 23%Z); ("nRW", Val_Symbolic 28%Z); ("N", Val_Symbolic 17%Z); ("SS", Val_Symbolic 21%Z); ("Q", Val_Symbolic 19%Z); ("GE", Val_Symbolic 11%Z); ("V", Val_Symbolic 26%Z); ("PAN", Val_Symbolic 18%Z); ("C", Val_Symbolic 5%Z); ("F", Val_Symbolic 10%Z)]) Mk_annot;
    ReadReg "CFG_ID_AA64PFR0_EL1_EL2" [] (Val_Bits [BV{4%N} 0x1%Z]) Mk_annot;
    ReadReg "CFG_ID_AA64PFR0_EL1_EL3" [] (Val_Bits [BV{4%N} 0x1%Z]) Mk_annot;
    ReadReg "CFG_ID_AA64PFR0_EL1_EL3" [] (Val_Bits [BV{4%N} 0x1%Z]) Mk_annot;
    ReadReg "__highest_el_aarch32" [] (Val_Bool false) Mk_annot;
    ReadReg "SCR_EL3" [] (Val_Bits [BV{32%N} 0x0%Z]) Mk_annot;
    ReadReg "CFG_ID_AA64PFR0_EL1_EL2" [] (Val_Bits [BV{4%N} 0x1%Z]) Mk_annot;
    ReadReg "CFG_ID_AA64PFR0_EL1_EL0" [] (Val_Bits [BV{4%N} 0x1%Z]) Mk_annot;
    ReadReg "CFG_ID_AA64PFR0_EL1_EL1" [] (Val_Bits [BV{4%N} 0x1%Z]) Mk_annot;
    ReadReg "CFG_ID_AA64PFR0_EL1_EL2" [] (Val_Bits [BV{4%N} 0x1%Z]) Mk_annot;
    ReadReg "CFG_ID_AA64PFR0_EL1_EL3" [] (Val_Bits [BV{4%N} 0x1%Z]) Mk_annot;
    ReadReg "__v81_implemented" [] (Val_Bool true) Mk_annot;
    ReadReg "__v84_implemented" [] (Val_Bool false) Mk_annot;
    ReadReg "CFG_ID_AA64PFR0_EL1_EL3" [] (Val_Bits [BV{4%N} 0x1%Z]) Mk_annot;
    ReadReg "CFG_ID_AA64PFR0_EL1_EL3" [] (Val_Bits [BV{4%N} 0x1%Z]) Mk_annot;
    ReadReg "__highest_el_aarch32" [] (Val_Bool false) Mk_annot;
    ReadReg "SCR_EL3" [] (Val_Bits [BV{32%N} 0x0%Z]) Mk_annot;
    ReadReg "TCR_EL2" [] (Val_Bits [BV{64%N} 0x0%Z]) Mk_annot;
    ReadReg "__v83_implemented" [] (Val_Bool false) Mk_annot;
    WriteReg "_PC" [] (Val_Symbolic 56%Z) Mk_annot;
    WriteReg "__PC_changed" [] (Val_Bool true) Mk_annot
  ];
  [
    Smt (DeclareConst 3%Z (Ty_BitVec 1%N)) Mk_annot;
    Smt (DeclareConst 4%Z (Ty_BitVec 2%N)) Mk_annot;
    Smt (DeclareConst 5%Z (Ty_BitVec 1%N)) Mk_annot;
    Smt (DeclareConst 6%Z (Ty_BitVec 1%N)) Mk_annot;
    Smt (DeclareConst 7%Z (Ty_BitVec 1%N)) Mk_annot;
    Smt (DeclareConst 8%Z (Ty_BitVec 1%N)) Mk_annot;
    Smt (DeclareConst 10%Z (Ty_BitVec 1%N)) Mk_annot;
    Smt (DeclareConst 11%Z (Ty_BitVec 4%N)) Mk_annot;
    Smt (DeclareConst 12%Z (Ty_BitVec 1%N)) Mk_annot;
    Smt (DeclareConst 13%Z (Ty_BitVec 1%N)) Mk_annot;
    Smt (DeclareConst 14%Z (Ty_BitVec 8%N)) Mk_annot;
    Smt (DeclareConst 15%Z (Ty_BitVec 1%N)) Mk_annot;
    Smt (DeclareConst 16%Z (Ty_BitVec 5%N)) Mk_annot;
    Smt (DeclareConst 17%Z (Ty_BitVec 1%N)) Mk_annot;
    Smt (DeclareConst 18%Z (Ty_BitVec 1%N)) Mk_annot;
    Smt (DeclareConst 19%Z (Ty_BitVec 1%N)) Mk_annot;
    Smt (DeclareConst 21%Z (Ty_BitVec 1%N)) Mk_annot;
    Smt (DeclareConst 22%Z (Ty_BitVec 1%N)) Mk_annot;
    Smt (DeclareConst 23%Z (Ty_BitVec 1%N)) Mk_annot;
    Smt (DeclareConst 24%Z (Ty_BitVec 1%N)) Mk_annot;
    Smt (DeclareConst 25%Z (Ty_BitVec 1%N)) Mk_annot;
    Smt (DeclareConst 26%Z (Ty_BitVec 1%N)) Mk_annot;
    Smt (DeclareConst 27%Z (Ty_BitVec 1%N)) Mk_annot;
    Smt (DeclareConst 28%Z (Ty_BitVec 1%N)) Mk_annot;
    Cycle Mk_annot;
    ReadReg "SEE" [] (Val_I (-1)%Z 128%Z) Mk_annot;
    WriteReg "SEE" [] (Val_I 1498%Z 128%Z) Mk_annot;
    WriteReg "__unconditional" [] (Val_Bool true) Mk_annot;
    ReadReg "__v85_implemented" [] (Val_Bool false) Mk_annot;
    ReadReg "PSTATE" [Field "Z"] (Val_Struct [("M", Val_Symbolic 16%Z); ("E", Val_Symbolic 8%Z); ("DIT", Val_Symbolic 7%Z); ("BTYPE", Val_Symbolic 4%Z); ("J", Val_Symbolic 15%Z); ("I", Val_Symbolic 12%Z); ("SP", Val_Bits [BV{1%N} 0x1%Z]); ("D", Val_Symbolic 6%Z); ("TCO", Val_Symbolic 24%Z); ("Z", Val_Symbolic 27%Z); ("A", Val_Symbolic 3%Z); ("IL", Val_Symbolic 13%Z); ("EL", Val_Bits [BV{2%N} 0x2%Z]); ("IT", Val_Symbolic 14%Z); ("SSBS", Val_Symbolic 22%Z); ("UAO", Val_Symbolic 25%Z); ("T", Val_Symbolic 23%Z); ("nRW", Val_Symbolic 28%Z); ("N", Val_Symbolic 17%Z); ("SS", Val_Symbolic 21%Z); ("Q", Val_Symbolic 19%Z); ("GE", Val_Symbolic 11%Z); ("V", Val_Symbolic 26%Z); ("PAN", Val_Symbolic 18%Z); ("C", Val_Symbolic 5%Z); ("F", Val_Symbolic 10%Z)]) Mk_annot;
    Smt (DefineConst 38%Z (Unop (Not) (Binop (Eq) (Val (Val_Symbolic 27%Z) Mk_annot) (Val (Val_Bits [BV{1%N} 0x1%Z]) Mk_annot) Mk_annot) Mk_annot)) Mk_annot;
    Branch 0%Z "model/aarch64.sail 12137:4 - 12139:5" Mk_annot;
    Smt (Assert (Unop (Not) (Val (Val_Symbolic 38%Z) Mk_annot) Mk_annot)) Mk_annot
  ]
].
